src/main/java/bio/overture/ego/config/SessionConfig.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) 2020. The Ontario Institute for Cancer Research. All rights reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package bio.overture.ego.config;
+
+import java.util.concurrent.ConcurrentHashMap;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.session.MapSessionRepository;
+import org.springframework.session.config.annotation.web.http.EnableSpringHttpSession;
+import org.springframework.session.web.http.CookieSerializer;
+import org.springframework.session.web.http.DefaultCookieSerializer;
+
+@EnableSpringHttpSession
+public class SessionConfig {
+
+  private final LoginNonceProperties properties;
+
+  public SessionConfig(@Autowired LoginNonceProperties properties) {
+    this.properties = properties;
+  }
+
+  /**
+   * Use in memory data store. TODO: Add support for redis and psql data store configurable behind
+   * profiles.
+   *
+   * @return SessionRepository implementation.
+   */
+  @Bean
+  public MapSessionRepository sessionRepository() {
+    return new MapSessionRepository(new ConcurrentHashMap<>());
+  }
+
+  @Bean
+  public CookieSerializer cookieSerializer() {
+    DefaultCookieSerializer serializer = new DefaultCookieSerializer();
+
+    // These can be user configured
+    serializer.setCookieName(properties.getName());
+    serializer.setSameSite(properties.getSameSite());
+    serializer.setUseSecureCookie(properties.isSecure());
+    serializer.setCookieMaxAge(properties.getMaxAge());
+
+    // These shouldn't be user configurable. ALWAYS make sure its HTTP Only.
+    serializer.setUseHttpOnlyCookie(true);
+    serializer.setCookiePath("/");
+    serializer.setDomainNamePattern("^.+?\\.(\\w+\\.[a-z]+)$");
```

Comment:
@andricDu  Plugging this into regex101.com, `example.com` does not match, but `anything.example.com` does. Not an issue for us as domain will have more than 2 parts, but just making sure this is what we want

Reply:
From the Spring Session docs: https://docs.spring.io/spring-session/docs/2.0.0.M2/reference/html5/guides/java-custom-cookie.html#custom-cookie-sample


> We customize the domain name pattern (a regular expression) to be ^.?\\.(\\w\\.[a-z]+)$ This allows sharing a session across domains and applications. If the regular expression does not match, no domain is set and the existing domain will be used. If the regular expression matches, the first grouping will be used as the domain. This means that a request to https://child.example.com will set the domain to example.com. However, a request to http://localhost:8080/ or http://192.168.1.100:8080/ will leave the cookie unset and thus still work in development without any changes necessary for production.

Reply:
Might as well make this configurable though. I can see someone locking down the allowed domains via a pattern.

Reply:
@rtisma I've moved it into the application.yml